save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/UserInfoAwareProps.kt

```diff
@@ -0,0 +1,55 @@
+/**
+ * Props for UserInfo
+ */
+
+package com.saveourtool.save.frontend.utils
+
+import com.saveourtool.save.info.UserInfo
+import react.FC
+import react.Props
+import react.PropsWithChildren
+import react.StateSetter
+
+/**
+ * Property to propagate user info from App
+ */
+external interface UserInfoAwareProps : Props {
+    /**
+     * Currently logged-in user or null
+     */
+    var userInfo: UserInfo?
+}
+
+/**
+ * Property to propagate user info from App with children
+ */
+external interface UserInfoAwarePropsWithChildren : UserInfoAwareProps, PropsWithChildren
+
+/**
+ * Property to propagate user info from App with ability to update it
+ */
+external interface UserInfoAwareMutableProps : UserInfoAwareProps {
+    /**
+     * Setter of user info (it can be updated in settings on several views)
+     *
+     * After updating user information we will update userSettings without re-rendering the page
+     * PLEASE NOTE: THIS PROPERTY AFFECTS RENDERING OF WHOLE APP.KT
+     * IF YOU HAVE SOME PROBLEMS WITH IT, CHECK THAT YOU HAVE PROPAGATED IT PROPERLY:
```

Comment:
Users of this code will definitely forget about it and will have hours of investigation where did they forgot to pass `this.setter = setter` in a call chain. No doubts, but anyway, lgtm